esp-hal-common/src/soc/esp32p4/peripherals.rs

```diff
@@ -0,0 +1,116 @@
+//! # Peripheral instance singletons (ESP32-P4)
+//!
+//! ## Overview
+//!
+//! The `Peripherals` module provides singleton instances of various peripherals
+//! and allows users to access and use them in their applications.
+//!
+//! These peripherals provide various functionalities and interfaces for
+//! interacting with different hardware components on the `ESP32-P4` chip, such
+//! as timers, `GPIO` pins, `I2C`, `SPI`, `UART`, and more. Users can access and
+//! utilize these peripherals by importing the respective singleton instances
+//! from this module.
+//!
+//! It's important to note that the module also exports the `Interrupt` enum
+//! from the `ESP32-P4` `PAC (Peripheral Access Crate)` for users to handle
+//! interrupts associated with these peripherals.
+//!
+//! ⚠️ NOTE: notice that `radio` and `lp_core` are marked with `false` in the
```

Comment:
Do we have the `radio` virtual peripheral in p4?

Reply:
It doesn't have any radios

Reply:
That comment is outdated, will jsut remove it